cdap-gateway/src/main/java/io/cdap/cdap/gateway/router/handlers/HttpRequestRouter.java

```diff
@@ -93,7 +94,7 @@ public HttpRequestRouter(CConfiguration cConf, RouterServiceLookup serviceLookup
   @Override
   public void channelRead(final ChannelHandlerContext ctx, final Object msg) throws Exception {
     try {
-      final Channel inboundChannel = ctx.channel();
+      this.inboundChannel = ctx.channel();
```

Comment:
This is not a correct channel. The HttpRequestRouter is a handler for Router<>IVP channel (inboundChannel). 
OutboundHandler is the handler for Router<>Service channel (this channel is named outboundChannel).
When you need is to stop reading from Router<>Service channel (set  outboundChannel.autoRead to false) when Router->IVP is full (not writable). So, you need to get outboundChannel from the currentMessageSender.